src/Integration.Vsix/CFamily/VcxProject/IFileInSolutionIndicator.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.ComponentModel.Composition;
+using EnvDTE;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.Integration.Vsix.CFamily.VcxProject;
+
+public interface IFileInSolutionIndicator
+{
+    bool IsFileInSolution(ProjectItem projectItem);
+}
+
+[Export(typeof(IFileInSolutionIndicator))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+[method:ImportingConstructor]
+public class FileInSolutionIndicator(IThreadHandling threadHandling) : IFileInSolutionIndicator
+{
+    public bool IsFileInSolution(ProjectItem projectItem)
+    {
+        threadHandling.ThrowIfNotOnUIThread();
+
+        try
+        {
+            // Issue 667:  https://github.com/SonarSource/sonarlint-visualstudio/issues/667
```

Comment:
Is this comment still relevant? 🤔 If yes it should be moved to the place where we actually create a vcxproject

Reply:
Still relevant, it's also in the correct place as it is about the check on line 49